src/game/Maps/Map.cpp

```diff
@@ -1260,6 +1260,62 @@ void Map::SetNewDifficultyCooldown(TimePoint const& newCooldown)
         m_dynamicDifficultyCooldown = newCooldown;
 }
 
+float Map::GetPetKillXPMod() const
```

Comment:
Expansion() Does this change with what expansion a map is on? I assumed this function returned the value from the account. If that is the case, as far as I can tell, this would only set the xp rate to the expansion setting in the account regardless of the map. I was under the impression that the desired behavior was to have variable rates for different expansion content (maps) ?
Perhaps this change would do what your trying to do?
```
float Map::GetPetKillXPMod() const
{    
    MapEntry const* mapEntry = sMapStore.LookupEntry(GetMapId());
    switch (mapEntry->Expansion())
    {
        default:
        case 0:
            return sWorld.getConfig(CONFIG_FLOAT_RATE_PET_XP_KILL_VANILLA);
        case 1:
            return sWorld.getConfig(CONFIG_FLOAT_RATE_PET_XP_KILL_BC);
        case 2:
            return sWorld.getConfig(CONFIG_FLOAT_RATE_PET_XP_KILL_WOTLK);
    }
} 
```

Reply:
`Expansion()` does indeed change with the expansion that a map is from. It does not reference the player's account; it does effectively the same thing as what you suggested. At map load time, `i_mapEntry` will have been fetched from `sMapStore`. Then, calling `Expansion()` on the map will return the expansion value that was stored in `Maps.dbc`. No point in re-querying the DBC, when we've already got a reference to the correct record.

Reply:
Player wont be allowed to enter expansion map if he has no rights for given expansion. So correct assumption. Doing a lookup here is meh.